src/jdk.compiler/share/classes/com/sun/tools/javac/code/Type.java

```diff
@@ -1943,12 +1943,34 @@ public TypeKind getKind() {
          */
         public Type lower;
 
+        public boolean universal = false;
```

Comment:
Can we use a flag on the tsym for this? After all, being "universal" is a declaration (symbol) not a use (type) property?

Reply:
yes, but `universal` is a property of the declaration, while `ref-ness` is a property of the use-site (hence type). Note that the underlying Valhalla compiler already has machinery to talk about reference projections, so perhaps we should use that to speak about `T` vs. `T.ref`, rather than inventing a new one. But we still need a flag somewhere to say whether the variable was declared as universal or not (as that affects what's compatible with that var).